homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,542 @@
+  expose_by_default is set to true. If not specified, this defaults to the
+  following list:
+      - 'switch'
+      - 'light'
+      - 'script'
+      - 'scene'
+      - 'group'
+      - 'input_boolean'
+      - 'media_player'
+
+The following are attributes that can be applied in the customize section:
+
+- echo specifies whether or not the entity should be discoverable by Alexa. The
+  default for this attribute is controlled with the expose_by_default config
+  option.
+
+- echo_friendly_name specifies the name that Alexa will know the entity as. The
+  default for this is the entity's friendly name.
+
+Much of this code is based on work done by Bruce Locke on his ha-local-echo
+project, located at "https://github.com/blocke/ha-local-echo", originally
+released under the MIT License. The license is reproduced below:
+
+The MIT License (MIT)
+
+Copyright (c) 2016 Bruce Locke
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+import threading
+import socket
+import logging
+import json
+
+import voluptuous as vol
+
+from homeassistant import util, core
+from homeassistant.const import (
+    ATTR_ENTITY_ID, SERVICE_TURN_OFF, SERVICE_TURN_ON,
+    EVENT_HOMEASSISTANT_START
+)
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_SUPPORTED_FEATURES, SUPPORT_BRIGHTNESS
+)
+import homeassistant.helpers.config_validation as cv
+
+DOMAIN = "alexa_local_control"
+
+REQUIREMENTS = ['Flask==0.11.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_HOST_IP = 'host_ip'
+CONF_LISTEN_PORT = 'listen_port'
+CONF_OFF_MAPS_TO_ON_DOMAINS = 'off_maps_to_on_domains'
+CONF_EXPOSE_BY_DEFAULT = 'expose_by_default'
+CONF_EXPOSED_DOMAINS = 'exposed_domains'
+
+ATTR_ECHO = 'echo'
+ATTR_ECHO_FRIENDLY_NAME = 'echo_friendly_name'
+
+DEFAULT_LISTEN_PORT = 8300
+DEFAULT_OFF_MAPS_TO_ON_DOMAINS = ['script', 'scene']
+DEFAULT_EXPOSE_BY_DEFAULT = True
+DEFAULT_EXPOSED_DOMAINS = [
+    'switch', 'light', 'script', 'scene', 'group', 'input_boolean',
+    'media_player'
+]

```

Comment:
Should `script` and `scene` both be in `DEFAULT_EXPOSED_DOMAINS` since they are already in `DEFAULT_OFF_MAPS_TO_ON_DOMAINS`? I would think that `DEFAULT_EXPOSED_DOMAINS` should only accept fully supported domains (everything that is already in the list mins `script` and `scene`).